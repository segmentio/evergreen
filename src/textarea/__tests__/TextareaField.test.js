import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { TextareaField } from '..'

const makeTextareaFieldFixture = (props = {}) => (
  <TextareaField data-testid="TextareaField" label="TextareaField" {...props} />
)

describe('TextareaField', () => {
  it('Should render', () => {
    expect(() => render(makeTextareaFieldFixture())).not.toThrow()
  })

  it('Should pass through `resize` prop to textarea', () => {
    const tree = renderer.create(makeTextareaFieldFixture({ resize: 'none' })).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
