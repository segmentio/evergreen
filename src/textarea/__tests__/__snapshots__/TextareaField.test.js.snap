// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaField Should pass through \`resize\` prop to textarea 1`] = `
<div
  className="ub-mb_24px ub-box-szg_border-box"
>
  <div
    className="ub-dspl_flex ub-flx-drct_column ub-mb_8px ub-box-szg_border-box"
  >
    <label
      className="ub-color_101840 ub-fnt-fam_b77syt ub-f-wght_500 ub-fnt-sze_14px ub-ln-ht_18px ub-ltr-spc_-0-05px ub-dspl_block ub-mb_0px ub-box-szg_border-box"
      htmlFor="TextareaField-2"
    >
      TextareaField
       
    </label>
  </div>
  <textarea
    aria-invalid={false}
    className="css-12rdf6g ub-w_100prcnt ub-h_80px ub-fnt-fam_b77syt ub-b-btm_1px-solid-transparent ub-b-lft_1px-solid-transparent ub-b-rgt_1px-solid-transparent ub-b-top_1px-solid-transparent ub-otln_iu2jf4 ub-txt-deco_none ub-min-h_80px ub-pl_12px ub-pr_12px ub-pb_8px ub-pt_8px ub-bblr_4px ub-bbrr_4px ub-btlr_4px ub-btrr_4px ub-ln-ht_16px ub-fnt-sze_12px ub-color_474d66 ub-tstn_n1akt6 ub-bg-clr_white ub-b-btm-clr_d8dae5 ub-b-lft-clr_d8dae5 ub-b-rgt-clr_d8dae5 ub-b-top-clr_d8dae5 ub-rsz_none ub-box-szg_border-box"
    data-gramm_editor={false}
    data-testid="TextareaField"
    disabled={false}
    id="TextareaField-2"
    spellCheck={true}
  />
</div>
`;
